tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,333 @@
+									EventType = "eventType.KeyValueDeleted",
+                                    SyncToken = null
+                                    },
+              new PushNotification  {
+                                    ResourceUri = null,
+                                    EventType = "eventType.KeyValueDeleted",
+                                    SyncToken = null
+                                    },
+              new PushNotification  {
+                                    ResourceUri = null,
+                                    EventType = null,
+                                    SyncToken = null
+                                    }
+            };
+
+		Dictionary<string, EventGridEvent> eventGridEvents = new Dictionary<string, EventGridEvent>
+		{
+            {
+                "sn;Vxujfidne",
+            new EventGridEvent(
+                "https://store1.resource.io/kv/searchQuery1",
+				"Microsoft.AppConfiguration.KeyValueModified", "2",
+                BinaryData.FromString("{\"key\":\"searchQuery1\",\"etag\":\"etagValue1\",\"syncToken\":\"sn;Vxujfidne\"}")
+                )
+            },
+
+            {
+                "sn;AxRty78B",
+            new EventGridEvent(
+                "https://store2.resource.io/kv/searchQuery1",
+                "Microsoft.AppConfiguration.KeyValueDeleted", "2",
+                BinaryData.FromString("{\"key\":\"searchQuery1\",\"etag\":\"etagValue1\",\"syncToken\":\"sn;AxRty78B\"}")
+                )
+            },
+
+            {
+                "sn;Ttylmable",
+            new EventGridEvent(
+                "https://store1.resource.io/kv/searchQuery2",
+				"Microsoft.AppConfiguration.KeyValueDeleted", "2",
+                BinaryData.FromString("{\"key\":\"searchQuery1\",\"etag\":\"etagValue1\",\"syncToken\":\"sn;Ttylmable\"}")
+                )
+            },
+
+            {
+                "sn;CRAle3342",
+            new EventGridEvent(
+                "https://store2.resource.io/kv/searchQuery2",
+				"Microsoft.AppConfiguration.KeyValueModified", "2",
+                BinaryData.FromString("{\"key\":\"searchQuery1\",\"etag\":\"etagValue1\",\"syncToken\":\"sn;CRAle3342\"}")
+                )
+            }
+        };
+
+		ConfigurationSetting FirstKeyValue => _kvCollection.First();
+
+        [Fact]
+        public void ValidatePushNotificationCreation()
+        {
+			foreach (KeyValuePair<string, EventGridEvent> eventGridAndSync in eventGridEvents)
+            {
+				string syncToken = eventGridAndSync.Key;
+				EventGridEvent eventGridEvent = eventGridAndSync.Value; 
+
+				eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
+
+                Assert.NotNull(pushNotification);
+                Assert.Equal(eventGridEvent.EventType, pushNotification.EventType);
+                Assert.Equal(new Uri(eventGridEvent.Subject), pushNotification.ResourceUri);
+				Assert.Equal(syncToken, pushNotification.SyncToken);
```

Comment:
You can use [Uri.OriginalString](https://docs.microsoft.com/en-us/dotnet/api/system.uri.originalstring?view=net-5.0)  to compare the strings, instead of creating a new Uri object.